include/nn/pdm/pdm_cpp.h

```diff
@@ -0,0 +1,466 @@
+   asm("GetPlayEvent__Q2_2nn3pdmFPiPQ3_2nn3pdm9PlayEventiT3");
+
+/**
+ * The max amount of \link PlayLog \endlink that can
+ * be written into the save.
+ *
+ * \param outMaxLength
+ * A pointer to write the size to. Must not be \c NULL.
+ *
+ * \return
+ * 0 on success.
+ */
+uint32_t
+GetPlayLogMaxLength(uint32_t* outMaxLength)
+   asm("GetPlayLogMaxLength__Q2_2nn3pdmFPi");
+
+/**
+ * Gets the amount of an users \link PlayLog \endlink entries.
+ *
+ * \param outLength
+ * A pointer to write the size to. Must not be \c NULL.
+ *
+ * \param userId
+ * A user ID from 1-12 to indicate the User
+ * 
+ * \return
+ * 0 on success.
+ */
+uint32_t
+GetPlayLogLength(uint32_t* outLength, uint32_t userId)
+   asm("GetPlayLogLength__Q2_2nn3pdmFPii");
+
+/**
+ * 
+ *
+ * \param outStart
+ * A pointer to write the start to. Must not be \c NULL.
+ *
+ * \param userId
+ * A user ID from 1-12 to indicate the User
+ * 
+ * \return
+ * 0 on success.
+ */
+uint32_t
+GetPlayLogStart(uint32_t* outStart, uint32_t userId)
+   asm("GetPlayLogStart__Q2_2nn3pdmFPii");
+
+/**
+ * Gets an array of \link PlayLog \endlink entries.
+ *
+ * \param outAmount
+ * A pointer to write the amount of entries that have been
+ * written to the array. Must not be \c NULL.
+ * May be smaller than the amount passed to the function.
+ * 
+ * \param outPlayLogs
+ * A pointer to write an array of \link PlayLog \endlink
+ * to. Must not be \c NULL.
+ *
+ * \param userId
+ * A user ID from 1-12 to indicate the User
+ * 
+ * \param amount
+ * The max amount of entries to be written to the array
+ * 
+ * \return
+ * 0 on success.
+ */
+uint32_t
+GetPlayLog(uint32_t* outAmount, PlayLog* outPlayLogs, uint32_t userId, uint32_t amount)
+   asm("GetPlayLog__Q2_2nn3pdmFPiPQ3_2nn3pdm7PlayLogiT3");
+
+/**
+ * Gets an array of \link PlayLog \endlink entries.
+ * 
+ * \param outPlayLogs
+ * A pointer to write an array of \link PlayLog \endlink
+ * to. Must not be \c NULL.
+ *
+ * \param userId
+ * A user ID from 1-12 to indicate the User
+ * 
+ * \param amount
+ * The max amount of entries to be written to the array
+ * 
+ * \return
+ * Either the amount of entries that have been
+ * written to the array or 0xFFFFFF on error.
+ * May be smaller than the amount passed to the function.
+ */
+uint32_t
+GetPlayLog(PlayLog* outPlayLogs, uint32_t userId, uint32_t amount)
+   asm("GetPlayLog__Q2_2nn3pdmFPQ3_2nn3pdm7PlayLogiT2");
+
+/**
+ * The max amount of \link PlayStats \endlink that can
+ * be written into the save.
+ *
+ * \param outMaxLength
+ * A pointer to write the size to. Must not be \c NULL.
+ *
+ * \return
+ * 0 on success.
+ */
+uint32_t
+GetPlayStatsMaxLength(uint32_t* outMaxLenght)
+   asm("GetPlayStatsMaxLength__Q2_2nn3pdmFPi");
+
+/**
+ * Gets the amount of an users \link PlayStats \endlink entries.
+ *
+ * \param outLength
+ * A pointer to write the size to. Must not be \c NULL.
+ *
+ * \param userId
+ * A user ID from 1-12 to indicate the User
```

Comment:
Do you think these user IDs are the same as `nn::act::SlotNo`? I believe they also start from 1 and go up from there. If they are the same, it'd be good to include `<nn/ac.h>` and use `nn::act::SlotNo` instead, so it's really clear how to map user accounts to slot numbers (and so folks browsing the documentation can click through to functions like `GetSlotNo`).

Reply:
I don't think they're the same.
userId is a uint32 and is checked to be less than 13.
SlotNo is a uchar and I didn't see any checks in the lib.

Reply:
No worries, we can always change it later.